app/controllers/products_controller.rb

```diff
@@ -1,2 +1,87 @@
 class ProductsController < ApplicationController
+  def index 
+    @products = Product.all
+    
+    @specialty = Product.sort_by_category("specialty")
+    @flower = Product.sort_by_category("flower")
+    @annual = Product.sort_by_category("annual")
+  end
+  
+  def show
+    product_id = params[:id]
+    @product = Product.find_by(id: params[:id])
+    
+    if @product.nil?
+      flash[:error] = "Invalid product."
+      redirect_to products_path
+      return
+    end   
+  end
+  
+  def new
+    @product = Product.new
+  end
+  
+  def create
+    @product = Product.new(product_params) 
+    
+    if @product.save
+      flash[:success] = "Product successfully created."
+      redirect_to product_path(@product.id)
+      return
+    else
+      flash[:warning] = "Can't create product."
+      redirect_to products_path
+      return
+    end
+  end 
+  
+  def edit
+    @product = Product.find_by(id: params[:id])
+    
+    if @product.nil?
+      flash[:warning] = "Can't edit, invalid product."
+      redirect_to products_path
+      return
+    end
+  end
+  
+  def update
+    @product= Product.find_by(id: params[:id])
+    
+    if @product.update(product_params)
+      flash[:success] = "Product successfully updated."
+      redirect_to product_path(@product.id)
+      return
+    else
+      flash[:warning] = "Can't update product."
+      render :edit
+      return
+    end
+  end
+  
+  def destroy
```

Comment:
Actually, I don't think we need a destroy action. A merchant can "retire a product from being sold" which will hide it from being viewed, but there isn't anything that says a product can/should be deleted. So instead of a typical "destroy" method, we'd have something that invokes 'update' and changes the status.

Reply:
Ok. I will work on that.